// @sanity/base/lib/components/ZIndexProvider.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useZIndex = useZIndex;
exports.ZIndexProvider = ZIndexProvider;

var _react = _interopRequireWildcard(require("react"));

var _sanityCssCustomProperties = _interopRequireDefault(require("sanity:css-custom-properties"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function getCustomCSSPropertyNumber(key) {
  var rawValue = _sanityCssCustomProperties.default[key];

  if (rawValue === undefined) {
    // eslint-disable-next-line no-console
    console.warn("getCustomCSSPropertyNumber: the custom CSS property \"".concat(key, "\" is not defined"));
    return undefined;
  }

  var value = parseInt(rawValue, 10);

  if (isNaN(value)) {
    // eslint-disable-next-line no-console
    console.warn("getCustomCSSPropertyNumber: the custom CSS property \"".concat(key, "\" is not a number"));
    return undefined;
  }

  return value;
}

var defaults = {
  /*
    used by
    - Navbar
  */
  navbar: getCustomCSSPropertyNumber('--zindex-navbar') || 200,
  navbarPopover: getCustomCSSPropertyNumber('--zindex-navbar-popover') || 500000,
  navbarDialog: getCustomCSSPropertyNumber('--zindex-navbar-dialog') || 500001,

  /*
    used by:
    - DefaultPane
  */
  pane: getCustomCSSPropertyNumber('--zindex-pane') || 100,

  /*
    used by:
    - DefaultPane
  */
  paneResizer: getCustomCSSPropertyNumber('--zindex-pane-resizer') || 150,

  /*
    used by:
    - EditItemFoldOut
    - Spinner
    - ConnectorsOverlay
    - tippy.css
    - BaseDateTimeInput
  */
  portal: getCustomCSSPropertyNumber('--zindex-portal') || 200,

  /*
    used by tooltip
  */
  popover: getCustomCSSPropertyNumber('--zindex-popover') || 200,

  /*
    used by google-maps-input
  */
  modal: getCustomCSSPropertyNumber('--zindex-modal') || 200,

  /*
    used for `movingItem` in:
    packages/@sanity/base/src/styles/layout/helpers.css
  */
  movingItem: getCustomCSSPropertyNumber('--zindex-moving-item') || 10000,

  /*
    used for shadow behind the navbar search, and behind sidemenu
  */
  drawershade: getCustomCSSPropertyNumber('--zindex-drawershade') || 1000000,

  /*
    used for snackbar
  */
  drawer: getCustomCSSPropertyNumber('--zindex-drawer') || 1000001,
  // NOT IN USE
  dropdown: getCustomCSSPropertyNumber('--zindex-dropdown') || 200,
  navbarFixed: getCustomCSSPropertyNumber('--zindex-navbar-fixed') || 1010,
  fullscreenEdit: getCustomCSSPropertyNumber('--zindex-fullscreen-edit') || 1050,
  popoverBackground: getCustomCSSPropertyNumber('--zindex-popover-background') || 1060,
  tooltip: getCustomCSSPropertyNumber('--zindex-tooltip') || 200,
  modalBackground: getCustomCSSPropertyNumber('--zindex-modal-background') || 2000,
  spinner: getCustomCSSPropertyNumber('--zindex-spinner') || 3000
};
var ZIndexContext = /*#__PURE__*/(0, _react.createContext)(defaults);

function useZIndex() {
  return (0, _react.useContext)(ZIndexContext);
}

function ZIndexProvider(_ref) {
  var children = _ref.children;
  return /*#__PURE__*/_react.default.createElement(ZIndexContext.Provider, {
    value: defaults
  }, children);
}
